src/language/pattern-types.md

```diff
@@ -59,6 +204,39 @@ case [a, b]: // ...
 case const [a, b]: // ...
 ```
 
+## Variable
+
+`var bar, String str, final int _`
+
+Variable patterns bind new variables to values that have been matched or destructured. 
+They usually occur as part of a [destructuring pattern][destructure] in order to
+capture a destructured value.
+
+The variables are in scope in a region of code that is only reachable when the
+pattern has matched.
```

Comment:
**issue:** Rephrase this. It's a little hard to follow. Maybe two sentences, one that establishes scope and one that covers pattern matching?

Reply:
This section wasn't changed as part of this PR, it was just moved. The reviews for the original PR were pretty stringent so I'm hesitant to make changes now. I also wouldn't know how to break it up since the scope is directly characterized by the match, and that's the point of the sentence.

Reply:
I agree, it's better to leave a majority of the focus on this PR on the reordering.